src/components/table/columns.ts

```diff
@@ -34,6 +39,32 @@ export class ColumnDefinitionFactory {
     }
 }
 
+/**
+ * Formats the header of the column
+ *
+ * @param {Column} column the configuration for the column
+ *
+ * @returns {string | HTMLElement} custom component that renders a column header
+ */
+export const formatHeader = (column: Column) => (): string | HTMLElement => {
+    const element = document.createElement('div');
+    element.innerText = column.title;
+    if (!column.headerComponent) {
+        return;
+    }
+
+    const menu = document.createElement(column.headerComponent.name);
+    menu.setAttribute('class', 'tabulator-header-menu-button');
+
+    const props = column.headerComponent.props;
+
+    forIn(props, (value, key) => (menu[key] = value));
```

Comment:
We should not have to depend on an external library for this. The following will have the same behaviour
```suggestion
    Object.assign(menu, props);
```